### PR TITLE
[CP-3004] Added toast component

### PR DESCRIPTION
**Pull request by mkurczewski**
JIRA Reference: [CP-3004]

### :memo: Description ️

-

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-3004]: https://appnroll.atlassian.net/browse/CP-3004?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/generic-view/feature/src/lib/generic-toasts.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import { ReduxRootState } from "Core/__deprecated__/renderer/store"
+import RecursiveLayout from "./recursive-layout"
+import { selectViewConfig } from "generic-view/store"
+import { createSelector } from "reselect"
+import { isEmpty } from "lodash"
+import styled from "styled-components"
+
+const selectToastsToRender = createSelector(selectViewConfig, (config) => {
+  return Object.entries(config || {})
+    .filter(([, { component }]) => ["toast"].includes(component))
```
Comment (dkarski) [inline libs/generic-view/feature/src/lib/generic-toasts.tsx:17]:
```suggestion
    .filter(([, { component }]) => component === "toast")
```

File: libs/generic-view/feature/src/lib/generic-toasts.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+interface Props {
+  viewKey: string
+}
+
+export const GenericToasts: FunctionComponent<Props> = ({ viewKey }) => {
+  const toastsToRender = useSelector((state: ReduxRootState) =>
+    selectToastsToRender(state, { viewKey })
+  )
+  const toastsToRenderDependency = JSON.stringify(toastsToRender)
```
Comment (dkarski) [inline libs/generic-view/feature/src/lib/generic-toasts.tsx:29]: Why is this here: const toastsToRenderDependency = JSON.stringify(toastsToRender)? Is it to work around the issue of array reference changes?

File: libs/generic-view/store/src/lib/toasts/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { createReducer } from "@reduxjs/toolkit"
+import { openToast, removeToast } from "./actions"
+
+export interface Toast {
+  key: string
+  opened?: boolean
```
Comment (dkarski) [inline libs/generic-view/store/src/lib/toasts/reducer.ts:11]:
```suggestion
```

File: libs/generic-view/ui/src/lib/interactive/toast/toast.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    } else {
+      animationTimeoutRef.current = setTimeout(() => {
+        setOpened(false)
+      }, toastAnimationDuration)
+    }
+    return () => {
+      clearTimeout(visibilityTimeoutRef.current)
+      clearTimeout(animationTimeoutRef.current)
+    }
+  }, [dispatch, toastAnimationDuration, toastVisibilityDuration, visible])
```
Comment (dkarski) [inline libs/generic-view/ui/src/lib/interactive/toast/toast.tsx:55]:
```suggestion
  }, [toastAnimationDuration, toastVisibilityDuration, visible])
```

**Review by dkarski — APPROVED** [review]
✅

I did a quick review and only had some minor comments. From me, it's approved—please include what you consider important and merge without additional checking, so we don't block the queue.

File: libs/generic-view/feature/src/lib/generic-toasts.tsx (hunk 2 above)
Comment (dkarski) [reply libs/generic-view/feature/src/lib/generic-toasts.tsx:29]: We can leave this topic for the next sync

Comment (mkurczewski) [reply libs/generic-view/feature/src/lib/generic-toasts.tsx:29]: Sure, let's discuss it later.
